File: binding/src/charts/libqcoloraxis.cpp
#include <QAbstractAxis>
#include <QColorAxis>
#include <QLinearGradient>
#include <QObject>
#include <QString>
#include <QByteArray>
#include <cstring>
#include <type_traits>
#include <qcoloraxis.h>
#include "libqcoloraxis.h"
#include "libqcoloraxis.hxx"

QColorAxis* QColorAxis_new() {
	 return new VirtualQColorAxis();
}

QColorAxis* QColorAxis_new2(QObject* parent) {
	 return new VirtualQColorAxis(parent);
}

libqt_string QColorAxis_Tr(const char* s) {
	QString _ret = QColorAxis::tr(s);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

int QColorAxis_Type(const QColorAxis* self) {
	return self->type();
}

void QColorAxis_SetMin(QColorAxis* self, double min) {
	self->setMin(min);
}

double QColorAxis_Min(const QColorAxis* self) {
	return self->min();
}

void QColorAxis_SetMax(QColorAxis* self, double max) {
	self->setMax(max);
}

double QColorAxis_Max(const QColorAxis* self) {
	return self->max();
}

void QColorAxis_SetRange(QColorAxis* self, double min, double max) {
	self->setRange(min, max);
}

void QColorAxis_SetTickCount(QColorAxis* self, int count) {
	self->setTickCount(count);
}

int QColorAxis_TickCount(const QColorAxis* self) {
	return self->tickCount();
}

void QColorAxis_SetSize(QColorAxis* self, const double size) {
	self->setSize(size);
}

double QColorAxis_Size(const QColorAxis* self) {
	return self->size();
}

void QColorAxis_SetGradient(QColorAxis* self, const QLinearGradient* gradient) {
	self->setGradient(*gradient);
}

QLinearGradient* QColorAxis_Gradient(const QColorAxis* self) {
	return new QLinearGradient(self->gradient());
}

void QColorAxis_SetAutoRange(QColorAxis* self, bool autoRange) {
	self->setAutoRange(autoRange);
}

bool QColorAxis_AutoRange(const QColorAxis* self) {
	return self->autoRange();
}

void QColorAxis_MinChanged(QColorAxis* self, double min) {
	self->minChanged(min);
}

void QColorAxis_Connect_MinChanged(QColorAxis* self, intptr_t slot) {
    void (*slotFunc)(QColorAxis*, double) = reinterpret_cast<void (*)(QColorAxis*, double)>(slot);
    QColorAxis::connect(self, &QColorAxis::minChanged, [self, slotFunc](qreal min) {
	slotFunc(self, min);
    });
}

void QColorAxis_MaxChanged(QColorAxis* self, double max) {
	self->maxChanged(max);
}

void QColorAxis_Connect_MaxChanged(QColorAxis* self, intptr_t slot) {
    void (*slotFunc)(QColorAxis*, double) = reinterpret_cast<void (*)(QColorAxis*, double)>(slot);
    QColorAxis::connect(self, &QColorAxis::maxChanged, [self, slotFunc](qreal max) {
	slotFunc(self, max);
    });
}

void QColorAxis_RangeChanged(QColorAxis* self, double min, double max) {
	self->rangeChanged(min, max);
}

void QColorAxis_Connect_RangeChanged(QColorAxis* self, intptr_t slot) {
    void (*slotFunc)(QColorAxis*, double, double) = reinterpret_cast<void (*)(QColorAxis*, double, double)>(slot);
    QColorAxis::connect(self, &QColorAxis::rangeChanged, [self, slotFunc](qreal min, qreal max) {
	slotFunc(self, min, max);
    });
}

void QColorAxis_TickCountChanged(QColorAxis* self, int tickCount) {
	self->tickCountChanged(tickCount);
}

void QColorAxis_Connect_TickCountChanged(QColorAxis* self, intptr_t slot) {
    void (*slotFunc)(QColorAxis*, int) = reinterpret_cast<void (*)(QColorAxis*, int)>(slot);
    QColorAxis::connect(self, &QColorAxis::tickCountChanged, [self, slotFunc](int tickCount) {
	slotFunc(self, tickCount);
    });
}

void QColorAxis_GradientChanged(QColorAxis* self, const QLinearGradient* gradient) {
	self->gradientChanged(*gradient);
}

void QColorAxis_Connect_GradientChanged(QColorAxis* self, intptr_t slot) {
    void (*slotFunc)(QColorAxis*, const QLinearGradient*) = reinterpret_cast<void (*)(QColorAxis*, const QLinearGradient*)>(slot);
    QColorAxis::connect(self, &QColorAxis::gradientChanged, [self, slotFunc](const QLinearGradient& gradient) {
	slotFunc(self, gradient);
    });
}

void QColorAxis_SizeChanged(QColorAxis* self, const double size) {
	self->sizeChanged(size);
}

void QColorAxis_Connect_SizeChanged(QColorAxis* self, intptr_t slot) {
    void (*slotFunc)(QColorAxis*, const double) = reinterpret_cast<void (*)(QColorAxis*, const double)>(slot);
    QColorAxis::connect(self, &QColorAxis::sizeChanged, [self, slotFunc](const qreal size) {
	slotFunc(self, size);
    });
}

void QColorAxis_AutoRangeChanged(QColorAxis* self, bool autoRange) {
	self->autoRangeChanged(autoRange);
}

void QColorAxis_Connect_AutoRangeChanged(QColorAxis* self, intptr_t slot) {
    void (*slotFunc)(QColorAxis*, bool) = reinterpret_cast<void (*)(QColorAxis*, bool)>(slot);
    QColorAxis::connect(self, &QColorAxis::autoRangeChanged, [self, slotFunc](bool autoRange) {
	slotFunc(self, autoRange);
    });
}

libqt_string QColorAxis_Tr2(const char* s, const char* c) {
	QString _ret = QColorAxis::tr(s, c);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

libqt_string QColorAxis_Tr3(const char* s, const char* c, int n) {
	QString _ret = QColorAxis::tr(s, c, n);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

// Base class handler implementation
int QColorAxis_QBaseType(const QColorAxis* self) {
	auto* vqcoloraxis = dynamic_cast<const VirtualQColorAxis*>(self);
	if (vqcoloraxis && vqcoloraxis->isVirtualQColorAxis) {
vqcoloraxis->setQColorAxis_Type_IsBase(true);
	return vqcoloraxis->type();
}
}

// Auxiliary method to allow providing re-implementation
void QColorAxis_OnType(const QColorAxis* self, intptr_t slot) {
	auto* vqcoloraxis = dynamic_cast<const VirtualQColorAxis*>(self);
	if (vqcoloraxis && vqcoloraxis->isVirtualQColorAxis) {
vqcoloraxis->setQColorAxis_Type_Callback(reinterpret_cast<VirtualQColorAxis::QColorAxis_Type_Callback>(slot));
}
}

void QColorAxis_Delete(QColorAxis* self) {
    delete self;
}

File: binding/include/charts/libqcoloraxis.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");%%_IMPORTLIBS_%%
%%_STRUCTDEFS_%%

/// https://doc.qt.io/qt-6/qcoloraxis.html
pub const qcoloraxis = struct {

    /// New constructs a new QColorAxis object.
    pub fn New() QtC.QColorAxis {
        return qtc.QColorAxis_new();
    }


    /// New2 constructs a new QColorAxis object.
    pub fn New2(parent: ?*anyopaque) QtC.QColorAxis {
        return qtc.QColorAxis_new2(@ptrCast(parent));
    }


    pub fn Tr(s: []const u8, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QColorAxis_Tr(@ptrCast(s));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qcoloraxis.Tr: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn Type(self: ?*anyopaque, ) i32 {
        return qtc.QColorAxis_Type(@ptrCast(self));
    }

    /// Allows for overriding the related default method
    pub fn OnType(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) i32) void {
        qtc.QColorAxis_OnType(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseType(self: ?*anyopaque, ) i32 {
        return qtc.QColorAxis_QBaseType(@ptrCast(self));
    }

    pub fn SetMin(self: ?*anyopaque, min: f64) void {
        qtc.QColorAxis_SetMin(@ptrCast(self), min);
    }

    pub fn Min(self: ?*anyopaque, ) f64 {
        return qtc.QColorAxis_Min(@ptrCast(self));
    }

    pub fn SetMax(self: ?*anyopaque, max: f64) void {
        qtc.QColorAxis_SetMax(@ptrCast(self), max);
    }

    pub fn Max(self: ?*anyopaque, ) f64 {
        return qtc.QColorAxis_Max(@ptrCast(self));
    }

    pub fn SetRange(self: ?*anyopaque, min: f64, max: f64) void {
        qtc.QColorAxis_SetRange(@ptrCast(self), min, max);
    }

    pub fn SetTickCount(self: ?*anyopaque, count: i32) void {
        qtc.QColorAxis_SetTickCount(@ptrCast(self), count);
    }

    pub fn TickCount(self: ?*anyopaque, ) i32 {
        return qtc.QColorAxis_TickCount(@ptrCast(self));
    }

    pub fn SetSize(self: ?*anyopaque, size: f64) void {
        qtc.QColorAxis_SetSize(@ptrCast(self), size);
    }

    pub fn Size(self: ?*anyopaque, ) f64 {
        return qtc.QColorAxis_Size(@ptrCast(self));
    }

    pub fn SetGradient(self: ?*anyopaque, gradient: ?*anyopaque) void {
        qtc.QColorAxis_SetGradient(@ptrCast(self), @ptrCast(gradient));
    }

    pub fn Gradient(self: ?*anyopaque, ) QtC.QLinearGradient {
        return qtc.QColorAxis_Gradient(@ptrCast(self));
    }

    pub fn SetAutoRange(self: ?*anyopaque, autoRange: bool) void {
        qtc.QColorAxis_SetAutoRange(@ptrCast(self), autoRange);
    }

    pub fn AutoRange(self: ?*anyopaque, ) bool {
        return qtc.QColorAxis_AutoRange(@ptrCast(self));
    }

    pub fn MinChanged(self: ?*anyopaque, min: f64) void {
        qtc.QColorAxis_MinChanged(@ptrCast(self), min);
    }

    pub fn OnMinChanged(self: ?*anyopaque, callback: *const fn (?*anyopaque, f64) callconv(.c) void) void {
        qtc.QColorAxis_Connect_MinChanged(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn MaxChanged(self: ?*anyopaque, max: f64) void {
        qtc.QColorAxis_MaxChanged(@ptrCast(self), max);
    }

    pub fn OnMaxChanged(self: ?*anyopaque, callback: *const fn (?*anyopaque, f64) callconv(.c) void) void {
        qtc.QColorAxis_Connect_MaxChanged(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn RangeChanged(self: ?*anyopaque, min: f64, max: f64) void {
        qtc.QColorAxis_RangeChanged(@ptrCast(self), min, max);
    }

    pub fn OnRangeChanged(self: ?*anyopaque, callback: *const fn (?*anyopaque, f64, f64) callconv(.c) void) void {
        qtc.QColorAxis_Connect_RangeChanged(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn TickCountChanged(self: ?*anyopaque, tickCount: i32) void {
        qtc.QColorAxis_TickCountChanged(@ptrCast(self), tickCount);
    }

    pub fn OnTickCountChanged(self: ?*anyopaque, callback: *const fn (?*anyopaque, i32) callconv(.c) void) void {
        qtc.QColorAxis_Connect_TickCountChanged(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn GradientChanged(self: ?*anyopaque, gradient: ?*anyopaque) void {
        qtc.QColorAxis_GradientChanged(@ptrCast(self), @ptrCast(gradient));
    }

    pub fn OnGradientChanged(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque) callconv(.c) void) void {
        qtc.QColorAxis_Connect_GradientChanged(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn SizeChanged(self: ?*anyopaque, size: f64) void {
        qtc.QColorAxis_SizeChanged(@ptrCast(self), size);
    }

    pub fn OnSizeChanged(self: ?*anyopaque, callback: *const fn (?*anyopaque, f64) callconv(.c) void) void {
        qtc.QColorAxis_Connect_SizeChanged(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn AutoRangeChanged(self: ?*anyopaque, autoRange: bool) void {
        qtc.QColorAxis_AutoRangeChanged(@ptrCast(self), autoRange);
    }

    pub fn OnAutoRangeChanged(self: ?*anyopaque, callback: *const fn (?*anyopaque, bool) callconv(.c) void) void {
        qtc.QColorAxis_Connect_AutoRangeChanged(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn Tr2(s: []const u8, c: []const u8, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QColorAxis_Tr2(@ptrCast(s), @ptrCast(c));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qcoloraxis.Tr2: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn Tr3(s: []const u8, c: []const u8, n: i32, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QColorAxis_Tr3(@ptrCast(s), @ptrCast(c), n);
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qcoloraxis.Tr3: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    /// Delete this object from C++ memory.
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.QColorAxis_Delete(@ptrCast(self));
    }
};
